tests/Integration/Behaviour/Features/Scenario/Product/add_virtual_product_file.feature

```diff
@@ -0,0 +1,61 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s product --tags virtual-product-file
+@reset-database-before-feature
+@virtual-product-file
+@clear-downloads-after-feature
+Feature: Add virtual product file from BO (Back Office).
+  As an employee I want to be able to add file of virtual product.
+
+  Scenario: I add virtual product file without limiting access days, download times and without expiration date
+    And I add product "product1" with following information:
+      | name       | en-US:puffin icon |
+      | is_virtual | true              |
+    And virtual product "product1" should not have a file
+    And product product1 type should be virtual
+    When I add new virtual product "product1" file "file1" with following details:
+      | display name | puffin-logo.png |
+      | file name    | app_icon.png    |
+    Then virtual product "product1" should have a file "file1" with following details:
+      | display name         | puffin-logo.png     |
+      | access days          | 0                   |
+      | download times limit | 0                   |
+      | expiration date      | 0000-00-00 00:00:00 |
+
+  Scenario: I add virtual product file with limited access days, downloads and expiration date
+    Given product product1 type should be virtual
+    When I add new virtual product "product1" file "file2" with following details:
```

Comment:
So it is allowed to add two files to the same product? Then there should be a list of virtual files
Or (as you commented in the handler) is a product supposed to have only one file? I guess this should be discussed with @PrestaShop/product-team 

However we shouldn't let us be too much influenced by `ProductDownload::getIdFromIdProduct` or `Product::getFilenameFromIdProduct` which tend to indicate product has only one ProductDownload

Reply:
oh Its a wrong test. I've  added a todo reminder in handler to assert file existence, but forgot it here :smile: 
As far as it seems **Product cannot have multiple files**